src/hotspot/cpu/aarch64/c1_LIRAssembler_aarch64.cpp

```diff
@@ -283,9 +283,9 @@ void LIR_Assembler::osr_entry() {
       }
 #endif
       __ ldr(r19, Address(OSR_buf, slot_offset + 0));
+      __ ldr(r20, Address(OSR_buf, slot_offset + 1*BytesPerWord));
       __ str(r19, frame_map()->address_for_monitor_lock(i));
-      __ ldr(r19, Address(OSR_buf, slot_offset + 1*BytesPerWord));
-      __ str(r19, frame_map()->address_for_monitor_object(i));
+      __ str(r20, frame_map()->address_for_monitor_object(i));
```

Comment:
I think it would be better to use `ldp` explicitly here, rather than relying on the assembler to do the merge.

Reply:
OK.